File: SoccerPlayerApi/SoccerPlayerApi/Dtos/Simulations/FrameSimulationCreateDto.cs
using SoccerPlayerApi.Entities.Frames;

namespace SoccerPlayerApi.Dtos.Simulations;

public class FrameSimulationCreateDto
{
    public int FrameId { get; set; }
    public Frame Frame { get; set; }

    public int AlgorithmId { get; set; }
    public AlgorithmDto Algorithm { get; set; }
    public List<AlgorithmParameterValueDto> Values { get; set; }
}

File: SoccerPlayerApi/clientapp/src/app/models/forecasts/FrameSimulationCreateDto.ts
import { AlgorithmParameterValueDto } from "./algorithms/algorithmParameterValueDto";

export class FrameSimulationCreateDto {
    frameId: number;
    algorithmId: number;
    values: AlgorithmParameterValueDto[];

    constructor(
        frameId: number,
        algorithmId: number,
        values: AlgorithmParameterValueDto[]
    ) {
        this.frameId = frameId;
        this.algorithmId = algorithmId;
        this.values = values;
    }
}
